Makefile

```diff
@@ -222,21 +212,27 @@ TEST_TARGETS := test-unit test-unit-amino test-unit-proto test-ledger-mock test-
 # Test runs-specific rules. To add a new test target, just add
 # a new rule, customise ARGS or TEST_PACKAGES ad libitum, and
 # append the new rule to the TEST_TARGETS list.
-
 test-unit: ARGS=-tags='cgo ledger test_ledger_mock norace'
 test-unit-amino: ARGS=-tags='ledger test_ledger_mock test_amino norace'
 test-ledger: ARGS=-tags='cgo ledger norace'
 test-ledger-mock: ARGS=-tags='ledger test_ledger_mock norace'
 test-race: ARGS=-race -tags='cgo ledger test_ledger_mock'
 test-race: TEST_PACKAGES=$(PACKAGES_NOSIMULATION)
-
 $(TEST_TARGETS): run-tests
 
+# check-* compiles and collects tests without running them
+# note: go test -c doesn't support multiple packages yet (https://github.com/golang/go/issues/15513)
+CHECK_TEST_TARGETS := test-unit test-unit-amino
+check-test-unit: test-unit
+check-test-unit-amino: test-unit-amino
+$(CHECK_TEST_TARGETS): EXTRA_ARGS=-run=none
```

Comment:
is this suppose to be inserted on `make test`?

```sh
cosmos-sdk on  marko/remove_toTM2 [!?] via 🐹 v1.15.3 on ☁️ markobaricevic3778@gmail.com 
❯ make test
go test -mod=readonly -tags='cgo ledger test_ledger_mock norace'  -run=none ./...
```

cc @alessio @robert-zaremba

Reply:
It introduces new jobs, but it doesn't change a way how you run tests. The new jobs are to compile and not runing any test.

Reply:
but the old command to run tests doesn't run tests so it does change how you run tests.

Reply:
@marbar3778 fix is [here](https://github.com/cosmos/cosmos-sdk/pull/7364/files#r512022058) - mind opening a PR?

Reply:
https://github.com/cosmos/cosmos-sdk/pull/7677